src/invokers.ts

```diff
@@ -89,7 +94,19 @@ export function makeResolverInvoker<T>(resolver: Resolver<T>) {
      */
     return function memberInvoker(ctx: any, ...rest: any[]) {
       const container: AwilixContainer = ctx.state.container
-      const resolved: any = container.build(resolver)
+      if (!container) {
+        throw new Error('Awilix container not found on Koa state object. Please ensure you use either scopePerRequest or attachContainer')
+      }
+
+      if (singleton) {
+        if (!resolved) {
+          resolved = container.build(resolver)
+        }
+      }
+      else {
+        resolved = container.build(resolver)
```

Comment:
Minor thing, but when using scoped, the last instantiated controller will be assigned to `resolved` which prevents it from being garbage collected until the next request comes in. Might not be a big deal, but could be solved with an intermediary variable.